zephyr/zephyr-7b-alpha/model/model.py

```diff
@@ -0,0 +1,106 @@
+from threading import Thread
+
+import torch
+from transformers import (
+    AutoModelForCausalLM,
+    AutoTokenizer,
+    GenerationConfig,
+    TextIteratorStreamer,
+)
+
+CHECKPOINT = "HuggingFaceH4/zephyr-7b-alpha"
+
+
+class Model:
+    def __init__(self, **kwargs):
+        self.tokenizer = None
+        self.model = None
+
+    def load(self):
+        self.model = AutoModelForCausalLM.from_pretrained(
+            CHECKPOINT,
+            torch_dtype=torch.float16,
+            device_map="auto",
+        )
+
+        self.tokenizer = AutoTokenizer.from_pretrained(
+            CHECKPOINT,
+            device_map="auto",
+            torch_dtype=torch.float16,
+        )
+
+    def preprocess(self, request: dict):
+        generate_args = {
+            "max_new_tokens": 512,
+            "temperature": 1.0,
+            "top_p": 0.95,
+            "top_k": 50,
+            "repetition_penalty": 1.0,
+            "no_repeat_ngram_size": 0,
+            "use_cache": True,
+            "do_sample": True,
+            "eos_token_id": self.tokenizer.eos_token_id,
+            "pad_token_id": self.tokenizer.pad_token_id,
+        }
+        if "max_tokens" in request.keys():
```

Comment:
same nit as above. totally optional, just slightly more maintainable.

Reply:
yeah this is way better, better to not propagate this everywhere